dojo/tools/meterian/parser.py

```diff
@@ -0,0 +1,139 @@
+                    false_p=False,
+                    duplicate=False,
+                    out_of_scope=False,
+                    impact=severity,
+                    static_finding=True,
+                    dynamic_finding=False,
+                    file_path="Manifest file",
+                    unique_id_from_tool=advisory['id'],
+                    tags=[language]
+                )
+
+                if 'cve' in advisory:
+                    if "N/A" != advisory["cve"]:
+                        finding.cve = advisory["cve"]
+
+                if "cwe" in advisory:
+                    finding.cwe = int(advisory["cwe"].replace("CWE-", ""))
+
+                mitigation_msg = "## Remediation\n"
+                safe_versions = dependency_report["safeVersions"]
+                if "latestPatch" in safe_versions:
+                    mitigation_msg += "Upgrade " + lib_name + " to version " + safe_versions["latestPatch"] + " or higher."
+                elif "latestMinor" in safe_versions:
+                    mitigation_msg += "Upgrade " + lib_name + " to version " + safe_versions["latestMinor"] + " or higher."
+                elif "latestMajor" in safe_versions:
+                    mitigation_msg += "Upgrade " + lib_name + " to version " + safe_versions["latestMajor"] + "."
+                else:
+                    mitigation_msg = "We were not able to provide a safe version for this library.\nYou should consider replacing this component as it could be an issue for the safety of your application."
+                finding.mitigation = mitigation_msg
+
+                references = ""
+                for link in advisory["links"]:
+                    ref_link = self.get_reference_url(link)
+                    if ref_link is not None:
+                        references += "- " + ref_link + "\n"
+                if references != "":
+                    finding.references = references
+
+                findings.append(finding)
+
+        return findings
+
+    def get_severity(self, advisory):
+        # Following the CVSS Scoring per https://nvd.nist.gov/vuln-metrics/cvss
+        if 'cvss' in advisory:
+            if advisory['cvss'] <= 3.9:
+                severity = "Low"
+            elif advisory['cvss'] >= 4.0 and advisory['cvss'] <= 6.9:
+                severity = "Medium"
+            elif advisory['cvss'] >= 7.0 and advisory['cvss'] <= 8.9:
+                severity = "High"
+            else:
+                severity = "Critical"
+        else:
+            if advisory["severity"] == "SUGGEST" or advisory["severity"] == "NA" or advisory["severity"] == "NONE":
+                severity = "Info"
+            else:
+                severity = advisory["severity"].title()
+
+        return severity
+
+    def uri_validator(self, url):
+        try:
+            result = urlparse(url)
+            return all([result.scheme, result.netloc, result.path])
+        except:
+            return False
```

Comment:
We use `hyperlink` module for all ULR related code.

https://defectdojo.github.io/django-DefectDojo/contributing/how-to-write-a-parser/#do-not-parse-urls-by-hand

Reply:
Right, I will use that module.